Shared/Samples/Show viewshed from geoelement in scene/ShowViewshedFromGeoelementInSceneView.swift

```diff
@@ -46,7 +46,7 @@ struct ShowViewshedFromGeoelementInSceneView: View {
 
 private extension ShowViewshedFromGeoelementInSceneView {
     /// The view model for the sample.
-    class Model: ObservableObject {
+    class Model: ObservableObject, @unchecked Sendable {
```

Comment:
This is particularly bad - the timer assumes the block is Sendable, which ask for the captured model to be Sendable.

Blindly conform the model to Sendable is a bad practice 🤮 , but since the only use of it being sendable is to allow a scheduled animation, is that OK to just disregard potential data races in it? Animation isn't very important after all.

Would like to hear your thoughts on this and the `FileNMEASentenceReader` type. Should I instead log issues to refactor those code?

Reply:
There isn't a simple solution here. I would suggest trying another method (instead of the block-based one).

Reply:
I've made the following changes:

- For the NMEA sentence reader, I move it onto Main Actor so the timer executes the block on main thread.
- For the Viewshed tank animation, I move it to the view level.

Let me know your thoughts.